api/test/enum.test.js

```diff
@@ -0,0 +1,284 @@
+      let str = ventilationTypeFrench.toString()
+      let characters = [...str]
+      expect(characters[0]).toEqual('(')
+      expect(characters[characters.length - 1]).toEqual(')')
+    })
+
+    it('returns a query object which returns a value from the test data', async () => {
+      let query = ventilationTypeFrench('Ventilateur-récupérateur de chaleur')
+      let result = await collection.findOne(query)
+      expect(result.houseId).toEqual(12149)
+    })
+  })
+
+  describe('ventilationAirFlowRateLps', () => {
+    it('returns a query object', () => {
+      expect(ventilationAirFlowRateLps(200)).toEqual({
+        evaluations: {
+          $elemMatch: {
+            ventilations: {
+              $elemMatch: {
+                airFlowRateLps: 200,
+              },
+            },
+          },
+        },
+      })
+    })
+
+    it('has a toString function that produces an eval friendly expression', () => {
+      let str = ventilationAirFlowRateLps.toString()
+      let characters = [...str]
+      expect(characters[0]).toEqual('(')
+      expect(characters[characters.length - 1]).toEqual(')')
+    })
+
+    it('returns a query object which returns a value from the test data', async () => {
+      let query = ventilationAirFlowRateLps(220)
+      let result = await collection.findOne(query)
+      expect(result.houseId).toEqual(12149)
+    })
+  })
+
+  describe('ventilationAirFlowRateCfm', () => {
+    it('returns a query object', () => {
+      expect(ventilationAirFlowRateCfm(200)).toEqual({
+        evaluations: {
+          $elemMatch: {
+            ventilations: {
+              $elemMatch: {
+                airFlowRateCfm: 200,
+              },
+            },
+          },
+        },
+      })
+    })
+
+    it('has a toString function that produces an eval friendly expression', () => {
+      let str = ventilationAirFlowRateLps.toString()
+      let characters = [...str]
+      expect(characters[0]).toEqual('(')
+      expect(characters[characters.length - 1]).toEqual(')')
+    })
+
+    it('returns a query object which returns a value from the test data', async () => {
+      let query = ventilationAirFlowRateCfm(466.1536)
+      let result = await collection.findOne(query)
+      expect(result.houseId).toEqual(12149)
+    })
+  })
+
+  describe('ventilationEfficiency', () => {
+    it('returns a query object', () => {
+      expect(ventilationEfficiency(200)).toEqual({
+        evaluations: {
+          $elemMatch: {
+            ventilations: {
+              $elemMatch: {
+                efficiency: 200,
+              },
+            },
+          },
+        },
+      })
+    })
+
+    it('has a toString function that produces an eval friendly expression', () => {
+      let str = ventilationEfficiency.toString()
+      let characters = [...str]
+      expect(characters[0]).toEqual('(')
+      expect(characters[characters.length - 1]).toEqual(')')
+    })
```

Comment:
I'm not sure what's the craic with Jest testing, but in pytest has this notion of parameterized tests, where you can run the same test with lots of different parameters. ([here's an example of testing for dates](https://github.com/alphagov/notifications-api/blob/fbe505a9916aa0f68f70193f51be3654665997cd/tests/app/dao/test_date_utils.py#L20-L31).) 

Is this possible with Jest? These `... eval friendly expression` tests look pretty boilerplate.

Reply:
Totally possible. You're right that it feels like these could be generated. Now that the pattern is clearer, I'll explore when I refactor.

Reply:
Sounds good.